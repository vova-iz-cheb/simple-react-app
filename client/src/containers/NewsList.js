import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

export const NewsList = props => {
  const news = props.news;

  if (news.length) {
    return (
      <ul className="news__list">
        {news.map(item => (
          <li key={item._id}>
            <article>
              <h2>
                <NavLink className="news__link" to={'/news/' + item._id}>
                  {item.title}
                </NavLink>
              </h2>
              <div>
                <p>{item.content}</p>
              </div>
            </article>
          </li>
        ))}
      </ul>
    );
  } else {
    return <p>К сожалению нет новостей.</p>;
  }
};

NewsList.propTypes = {
  news: PropTypes.array.isRequired,
};
