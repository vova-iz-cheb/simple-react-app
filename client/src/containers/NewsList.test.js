import React from 'react';
import { NewsList } from './NewsList';
import { shallow } from 'enzyme';

describe('News List with 2 articles', () => {
  const news = [
    {
      _id: 1,
      title: 'Title',
      content: 'Content',
    },
    {
      _id: 2,
      title: 'Title2',
      content: 'Content2',
    },
  ];

  const newsList = shallow(<NewsList news={news} />);

  it('Snapshot', () => {
    expect(newsList).toMatchSnapshot();
  });

  it('Render initial', () => {
    console.log(newsList.debug());
    expect(newsList.find('ul.news__list h2')).toHaveLength(2);
    expect(newsList.find('ul.news__list p')).toHaveLength(2);

    expect(
      newsList
        .find('.news__link')
        .at(0)
        .text()
    ).toEqual('Title');

    expect(
      newsList
        .find('.news__link')
        .at(1)
        .text()
    ).toEqual('Title2');

    expect(
      newsList
        .find('ul.news__list p')
        .at(0) // equal first()
        .text()
    ).toEqual('Content');

    expect(
      newsList
        .find('ul.news__list p')
        .at(1)
        .text()
    ).toEqual('Content2');
  });
});

describe('empty NewsList', () => {
  const news = [];

  const newsList = shallow(<NewsList news={news} />);

  it('Snapshot', () => {
    expect(newsList).toMatchSnapshot();
  });

  it('Render initial', () => {
    console.log(newsList.debug());
    expect(newsList.find('ul.news__list h2')).toHaveLength(0);
    expect(newsList.find('ul.news__list p')).toHaveLength(0);
    expect(newsList.find('p').text()).toEqual('К сожалению нет новостей.');
  });
});
