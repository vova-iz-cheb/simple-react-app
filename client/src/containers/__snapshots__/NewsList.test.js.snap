// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News List with 2 articles Snapshot 1`] = `
<ul
  className="news__list"
>
  <li
    key="1"
  >
    <article>
      <h2>
        <NavLink
          className="news__link"
          to="/news/1"
        >
          Title
        </NavLink>
      </h2>
      <div>
        <p>
          Content
        </p>
      </div>
    </article>
  </li>
  <li
    key="2"
  >
    <article>
      <h2>
        <NavLink
          className="news__link"
          to="/news/2"
        >
          Title2
        </NavLink>
      </h2>
      <div>
        <p>
          Content2
        </p>
      </div>
    </article>
  </li>
</ul>
`;

exports[`empty NewsList Snapshot 1`] = `
<p>
  К сожалению нет новостей.
</p>
`;
